src/SumProc.jl

```diff
@@ -107,36 +108,45 @@ function nesum(a::CycloFrac{T}, var::Int64, lower::Int64, upper::Union{Int64,T},
 		sum, exc = nesum(a, var, 0, upper, congruence)
 		return (sum-nesum(a, var, 0, lower-1), exc)
 	end
+	# From now on `lower` can be assumed to be zero.
 	if congruence !== nothing
 		ring=parent(a.numerator.summands[1].modulus)
 		c=congruence[2]*gen(ring)+congruence[1]
 		cinv=(gen(ring)-congruence[1])//congruence[2]
 	end
 	sum=0
 	exceptions=Set{ParameterException{T}}()
-	for root in a.numerator.summands
-		if degrees(root.argument)[var] > 0
-			cl=coeff(root.argument, [var], [1])
-			co=eesubs(root.argument, [var], [0])
-			ke=e2p(cl)
-			summand=(root.modulus*e2p((upper+1)*cl+co)-root.modulus*e2p(co))//((ke-1)*a.denominator)
+	# Using the commutativity of the addition in the ring of generic cyclotomics
+	# the sum can be computed separately for each of the summands of `a.numerator`.
+	# Also `a.denominator` will be ignored and added back to the result later.
+	for summand in a.numerator.summands
+		var_degree=degrees(summand.argument)[var]
+		if var_degree <= 0  # `summand` doesn't depend on `var` at all and thus the sum boils down to a simple multiplication.
+			sum+=(upper+1)*summand
+		elseif isone(var_degree)  # `summand.argument` linearly depends on `var` hence the sum is a geometric sum.
+			var_coeff=coeff(summand.argument, [var], [1])
+			constant=eesubs(summand.argument, [var], [0])
+			geometric_sum=summand.modulus*e2p(constant)*(e2p((upper+1)*var_coeff)-1)//(e2p(var_coeff)-1)
 			if congruence === nothing
-				sum+=summand
+				sum+=geometric_sum
 			else
-				sum+=simplify(summand, c, cinv)
+				sum+=simplify(geometric_sum, c, cinv)
 			end
-			if !(ishalf(root.argument) && isunitfraction(cl))
+			# If `var_coeff` evaluates to an integer `e2p(var_coeff)` evaluates to one and thus `e2p(var_coeff)-1` to zero.
+			# So in this case the closed formular for the geometric sum doesn't hold.
+			if !(ishalf(var_coeff) && isunitfraction(var_coeff))  # TODO make this more general to skip more exceptions?
```

Comment:
We just discussed this code and after reading the code if `ishalf` and `isunitfraction` it seems as if this check says: "if `var_coeff` is 1 or 1/2 then DO NOT add parameter exceptions" -- but if I understand @SoongNoonien right, then it should really only exclude this for "1/2" ??   I guess the function `ishalf` is a bit confusing (and perhaps needs a better name)

Reply:
Ok, I've checked again and it seems like `var_coeff` can't be 1. `summand.argument` is reduced and doesn’t have any integral parts. So something like `summand.argument == 1//q*i+j` (with `var == j` this would mean `var_coeff == 1`) is not possible this would always be simplified to `1//q*i`. This is why we added the comment in line 33. But this comment is rather confusing.